FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/ContainerRegistry/AzureContainerRegistryAccessTokenProvider.cs

```diff
@@ -0,0 +1,184 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Synapse.DataClient.Api;
+using Microsoft.Health.Fhir.Synapse.SchemaManagement.Exceptions;
+using Newtonsoft.Json;
+using Polly;
+
+namespace Microsoft.Health.Fhir.Synapse.SchemaManagement.ContainerRegistry
+{
+    /// <summary>
+    /// Retrieve ACR access token with AAD token provider.
+    /// We need to exchange ACR refresh token with AAD token, and get ACR access token from refresh token.
+    /// References:
+    /// https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#calling-post-oauth2exchange-to-get-an-acr-refresh-token
+    /// https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#calling-post-oauth2token-to-get-an-acr-access-token
+    /// </summary>
+    public class AzureContainerRegistryAccessTokenProvider : IContainerRegistryTokenProvider
```

Comment:
If this class is similar to AzureContainerRegistryAccessTokenProvider in fhir server lib, can we try to leverage that class. If not, can we consider using a different class name?

Reply:
Updated to **ContainerRegistryAccessTokenProvider**